locust/contrib/fasthttp.py

```diff
@@ -0,0 +1,383 @@
+                )
+            return response
+    
+    def delete(self, path, **kwargs):
+        return self.request("DELETE", path, **kwargs)
+    
+    def get(self, path, **kwargs):
+        """Sends a GET request"""
+        return self.request("GET", path, **kwargs)
+    
+    def head(self, path, **kwargs):
+        """Sends a HEAD request"""
+        return self.request("HEAD", path, **kwargs)
+    
+    def options(self, path, **kwargs):
+        """Sends a OPTIONS request"""
+        return self.request("OPTIONS", path, **kwargs)
+    
+    def patch(self, path, data=None, **kwargs):
+        """Sends a POST request"""
+        return self.request("PATCH", path, data=data, **kwargs)
+    
+    def post(self, path, data=None, **kwargs):
+        """Sends a POST request"""
+        return self.request("POST", path, data=data, **kwargs)
+    
+    def put(self, path, data=None, **kwargs):
+        """Sends a PUT request"""
+        return self.request("PUT", path, data=data, **kwargs)
+
+
+class FastResponse(CompatResponse):
+    headers = None
+    """Dict like object containing the response headers"""
+    
+    _response = None
+    
+    @property
+    def text(self):
+        """
+        Returns the text content of the response as a decoded string
+        (unicode on python2)
+        """
+        # Decode unicode from detected encoding.
+        try:
+            content = str(self.content, self.apparent_encoding, errors='replace')
+        except (LookupError, TypeError):
+            # A LookupError is raised if the encoding was not found which could
+            # indicate a misspelling or similar mistake.
+            #
+            # A TypeError can be raised if encoding is None
+            #
+            # Fallback to decode without specifying encoding
+            content = str(self.content, errors='replace')
+        return content
+    
+    @property
+    def apparent_encoding(self):
+        """The apparent encoding, provided by the chardet library."""
+        return chardet.detect(self.content)['encoding']
+    
+    def raise_for_status(self):
+        """Raise any connection errors that occured during the request"""
+        if hasattr(self, 'error') and self.error:
+            raise self.error
+    
+    @property
+    def status_code(self):
+        """
+        We override status_code in order to return None if no valid response was 
+        returned. E.g. in the case of connection errors
+        """
+        return self._response is not None and self._response.get_code() or 0
+    
+    def _content(self):
+        if self.headers is None:
+            return None
+        return super(FastResponse, self)._content()
+
+
+class ErrorResponse(object):
+    """
+    This is used as a dummy response object when geventhttpclient raises an error 
+    that doesn't have a real Response object attached. E.g. a socket error or similar
+    """
+    headers = None
+    content = None
+    status_code = 0
```

Comment:
will a status code of 0 confuse people?

Reply:
The reason for setting status_code to 0 is that there are some case where Requests does this (I don't remember exactly when), and I wanted to keep the API as similar as possible.